api/tools/src/main/java/org/pmiops/workbench/tools/cdrConfig/UpdateCdrConfig.java

```diff
@@ -0,0 +1,331 @@
+package org.pmiops.workbench.tools.cdrConfig;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.pmiops.workbench.db.dao.AccessTierDao;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbAccessTier;
+import org.pmiops.workbench.db.model.DbCdrVersion;
+import org.pmiops.workbench.model.ArchivalStatus;
+import org.pmiops.workbench.tools.CommandLineToolConfig;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * See api/project.rb update-cdr-config. Reads Access Tiers and CDR versions from a JSON file and
+ * updates the database to match.
+ */
+@Configuration
+public class UpdateCdrConfig {
+
+  private static final Logger logger = Logger.getLogger(UpdateCdrConfig.class.getName());
+
+  @Bean
+  @Transactional
+  public CommandLineRunner run(
+      AccessTierDao accessTierDao, CdrVersionDao cdrVersionDao, WorkspaceDao workspaceDao)
+      throws IOException {
+    return (args) -> {
+      if (args.length != 2) {
+        throw new IllegalArgumentException(
+            "Expected 2 args (file, dry_run). Got " + Arrays.asList(args));
+      }
+
+      // it's small and we're parsing it multiple times, so read it into memory
+      final String configFile = new String(Files.readAllBytes(Paths.get(args[0])));
+      boolean dryRun = Boolean.parseBoolean(args[1]);
+
+      final Map<Long, Long> accessTierMap = generateAccessTierMap(configFile);
+
+      // no adapters necessary - Gson can infer the structure of DbAccessTier
+
+      final Gson accessTiersGson = new GsonBuilder().create();
+      final AccessTierExtractor accessTiers =
+          accessTiersGson.fromJson(configFile, AccessTierExtractor.class);
+
+      // DbCdrVersion requires Gson adapters for Timestamp formatting and DbAccessTier resolution
+
+      final Gson cdrVersionsGson =
+          new GsonBuilder()
+              .registerTypeAdapter(Timestamp.class, new TimestampGsonAdapter())
+              .registerTypeAdapter(DbAccessTier.class, new DbAccessTierIdGsonAdapter(accessTierDao))
```

Comment:
I'm not sure I follow how this is working on the first run. Wouldn't you need to have inserted the access tiers before you can use this parser? I guess perhaps this is why the gson parsing is deferred into later in the method.